src/Domain/MetaCampaign.php

```diff
@@ -251,6 +266,23 @@ public function getStatus(): ?string
     {
         return $this->status;
     }
+    
+    public function getStatusAt(\DateTimeImmutable $date): ?string
+    {
```

Comment:
port of Apex code from https://github.com/thebiggive/salesforce/blob/eef73c195e06f7365988b65b28095e4d23138c63/force-app/main/default/classes/CampaignService.cls#L217